sbncode/Calibration/TrackCaloSkimmer_module.cc

```diff
@@ -1400,13 +1457,29 @@ sbn::TrackHitInfo sbn::TrackCaloSkimmer::MakeHit(const recob::Hit &hit,
     }
   }
 
-  // This is needed to reconstrut drift coordinate using a different Time
-  int driftDir = geo->TPC(hit.WireID()).DriftDir().X();
-  const double driftv(dprop.DriftVelocity(dprop.Efield(), dprop.Temperature()));
-  double anodeDistance = std::numeric_limits<float>::signaling_NaN();
-  if(t0Info.hasT0CRTHit) anodeDistance = (hit.PeakTime()-dclock.Time2Tick(dclock.TriggerTime())-t0Info.t0CRTHit*1e-3/dclock.TPCClock().TickPeriod())*dclock.TPCClock().TickPeriod()*driftv;
-  double wirePlaneX = wireReadout->Plane(geo::PlaneID(hit.WireID().Cryostat, hit.WireID().TPC, hit.WireID().Plane)).GetCenter().X();
-  double recoX = wirePlaneX - driftDir*anodeDistance;
+  // If T0 provided CRT then we should appropriately shift the drift position (already done for pandora T0)
+  double recoX = std::numeric_limits<float>::signaling_NaN();
+
+  if(!t0Info.hasT0Pandora && (t0Info.hasT0CRTTrack || t0Info.hasT0CRTHit || t0Info.hasT0CRTSpacePoint))
+    {
+      int driftDir = geo->TPC(hit.WireID()).DriftDir().X();
+      const double driftv(dprop.DriftVelocity(dprop.Efield(), dprop.Temperature()));
+
+      double time = std::numeric_limits<float>::signaling_NaN();
+
+      if(t0Info.hasT0CRTTrack)
+        time = t0Info.t0CRTTrack*1e-3;
+      else if(t0Info.hasT0CRTHit)
+        time = t0Info.t0CRTHit*1e-3;
+      else if(t0Info.hasT0CRTSpacePoint)
+        time = t0Info.t0CRTSpacePoint*1e-3;
+
+      double anodeDistance = (hit.PeakTime()-dclock.Time2Tick(dclock.TriggerTime())-time/dclock.TPCClock().TickPeriod())*dclock.TPCClock().TickPeriod()*driftv;
+      double wirePlaneX = wireReadout->Plane(geo::PlaneID(hit.WireID().Cryostat, hit.WireID().TPC, hit.WireID().Plane)).GetCenter().X();
+
+      recoX = wirePlaneX - driftDir*anodeDistance;
```

Comment:
Since xshift is now being passed as a parameter, shouldn't you just apply it instead of calculating recoX here?

Backing up -- tp.x and sp.x should be shifted by the same amount. They shouldn't have different calculations.

Reply:
@gputnam @francescopoppi I am seeing ~0mm, ~3mm and ~6mm differences between this technique and pure use of the xshift. Clearly one of the methods doesn't / double accounts for interplane drift time.

I will try and investigate this this afternoon but given you implemented it I would welcome your input - do you know for example, whether this is over corrected?

Reply:
But clearly one of these instances is wrong so it's worth understanding which.

I would expect that the interplane drift is already accounted for in the creation of the spacepoints given they're already 3D constructs. I'm not familiar with the trajectory points so I would need to understand how they work.

Reply:
I have looked into this more - SPs explicitly account for the interplane drift as would be expected, TPs are based on the trajectory which itself is a fit to the SPs.

Correcting for interplane drift here is wrong. I've made a couple of plots that show that the SP x's give and agreed x range (with the first induction location being the limit) whilst the TP x's with their current correction give shifted distributions.
<img width="1200" height="800" alt="hits_all_tpc_0_sp_x" src="https://github.com/user-attachments/assets/37a00357-6157-475b-9a94-51629f4b1984" />
<img width="1200" height="800" alt="hits_all_tpc_0_tp_x" src="https://github.com/user-attachments/assets/c7e5f594-fddd-4383-8984-b26bf8965c98" />

Reply:
I am therefore going to remove this section and correct all locations by the xshift calculated previously, avoiding double correcting.

Reply:
Commit now pushed.

@mrmooney @linyan-w just making you aware of this too!